balancer-js/src/modules/data/pool/subgraph.ts

```diff
@@ -141,7 +141,7 @@ export class PoolsSubgraphRepository
 
   async findBy(param: PoolAttribute, value: string): Promise<Pool | undefined> {
     if (!this.pools) {
-      this.pools = this.fetchDefault();
+      this.pools = this.fetch();
```

Comment:
I'm not really sure about the implications of updating this. 
I can look deeper into this tomorrow, but if anyone already knows that this may cause issues, please let me know.

Reply:
I think one immediate issue is it will only fetch the first 1000 pools and I think we have >1000 on mainnet now (and definitely on Polygon)

Reply:
I'll rewrite to use AllPools instead of Pools, so we keep the same behavior as before